ros2/crystal/ubuntu/bionic/ros-core/Dockerfile

```diff
@@ -0,0 +1,45 @@
+# This is an auto generated Dockerfile for ros2:ros-core
+# generated from docker_images/create_ros2_core_image.Dockerfile.em
+FROM ubuntu:bionic
+
+RUN echo 'Etc/UTC' > /etc/timezone && \
+    ln -s /usr/share/zoneinfo/Etc/UTC /etc/localtime && \
+    apt-get update && apt-get install -q -y tzdata && rm -rf /var/lib/apt/lists/*
+
+
+# install packages
+RUN apt-get update && apt-get install -q -y \
+    dirmngr \
+    gnupg2 \
+    lsb-release \
+    python3-pip \
+    && rm -rf /var/lib/apt/lists/*
+
+
+# setup ros2 keys
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116
+
+# setup sources.list
+RUN . /etc/os-release \
+    && echo "deb http://repo.ros2.org/$ID/main `lsb_release -sc` main" > /etc/apt/sources.list.d/ros2-latest.list
+
+# setup environment
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+ENV ROS_DISTRO crystal
```

Comment:
Do you think we should install and init rosdep in the core ros2 image, like we do for ros1? E.g:
```
# install bootstrap tools
RUN apt-get update && apt-get install --no-install-recommends -y \
    python3-rosdep \
    python3-rosinstall \
    python3-vcstools \
    && rm -rf /var/lib/apt/lists/*

# bootstrap rosdep
ENV ROSDISTRO_INDEX_URL https://raw.githubusercontent.com/ros2/rosdistro/ros2/index.yaml
RUN rosdep init \
    && rosdep update
```
Also, do we still need to set the `ROSDISTRO_INDEX_URL`? Ping @tfoote

Reply:
> Do you think we should install and init rosdep in the core ros2 image, like we do for ros1?

I don't have a considered opinion on this but if we're doing it in the ROS images it's reasonable for end users to expect it in the ROS 2 images as well. ROS 2 has used the same rosdep database as ROS since at least Bouncy and maybe even Ardent, my memory fails me here. So adding the rosdep initialization should just be a net positive.



> Also, do we still need to set the `ROSDISTRO_INDEX_URL`?

ROS 2 is still using a forked rosdistro index in https://github.com/ros2/rosdistro. The documentation for [releasing a ROS 2 package with Bloom](https://index.ros.org/doc/ros2/Releasing-a-ROS-2-package-with-bloom/#differences-from-ros-1-bloom) will always have the latest information but for Crystal the ROSDISTRO_INDEX_URL should be set to
```
https://raw.githubusercontent.com/ros2/rosdistro/ros2/index-v4.yaml
```

Reply:
Is this `ROSDISTRO_INDEX_URL` expected to bump version often, e.g. I should parameterize it into the yaml config rather than leave it in the template.em? Can ardent/bounce/crystal all use the same index url from above? Will this index be roll back upstream from the `rosdistro` is forked, or indefinitely remain separate/independent?

Reply:
> Is this ROSDISTRO_INDEX_URL expected to bump version often, e.g. I should parameterize it into the yaml config rather than leave it in the template.em?

Hopefully this should be a highly infrequent event. The change is related to https://github.com/ros-infrastructure/rep/pull/181 for other images, the default rosdistro index url inside rosdep has already been updated to use the v4 index in ros/rosdistro.

> Can ardent/bounce/crystal all use the same index url from above?

Yep.

Reply:
Thanks, PR updated to incorporate bootstrapping tools.